modules/ml/src/main/java/org/apache/ignite/ml/structures/partition/LabeledDatasetPartitionDataBuilderOnHeap.java

```diff
@@ -57,8 +58,10 @@ public LabeledDatasetPartitionDataBuilderOnHeap(IgniteBiFunction<K, V, Vector> x
     }
 
     /** {@inheritDoc} */
-    @Override public LabeledVectorSet<Double, LabeledVector> build(Iterator<UpstreamEntry<K, V>> upstreamData,
-                                                                   long upstreamDataSize, C ctx) {
+    @Override public LabeledVectorSet<Double, LabeledVector> build(
+        LearningEnvironment env,
```

Comment:
maybe we move set of env through with.. method?

Reply:
No, we can do such things with entry point of environment in the chain of learning, but in the middle of it we should explicitly require `env` as a parameter. Imagine if we passed some learning environment builder in trainer, but forgot to set it in the middle of chain: it will be inconsistent chain.

Reply:
No, i think. Approach `with` + sensible default is good in the beginning of dependency chain, but in the middle of it we should force explicit passing of parameter to not break dependency because of forgetfulness to call set or sth like this. In our case beginning of chain is `DatasetTrainer`, and there we use this approah.